spec/controllers/reservations_controller_spec.rb

```diff
@@ -453,58 +404,57 @@
   end
 
   describe '#edit (GET /reservations/:id/edit)' do
-    ## Unhappy paths
+    # SMELLS:
+    #   - message chain
     it_behaves_like 'inaccessible by banned user' do
-      before { get 'edit', id: @reservation.id }
+      before { get 'edit', id: 1 }
     end
 
     context 'when accessed by patron' do
       before(:each) do
-        sign_in @user
-        get 'edit', id: @reservation.id
+        mock_user_sign_in
+        get 'edit', id: 1
       end
-      include_examples 'cannot access page'
+      include_examples 'redirected request'
     end
 
     context 'when accessed by checkout person disallowed by settings' do
       before(:each) do
-        sign_in @checkout_person
-        allow(@ac).to receive(:checkout_persons_can_edit).and_return(false)
-        get 'edit', id: @reservation.id
```

Comment:
Oh I just meant using a constant to hold the value of 1 (Or any other
arbitrary constant number)

On Aug 30, 2016 11:34 AM, "Sydney Young" <notifications@github.com> wrote:

> In spec/controllers/reservations_controller_spec.rb
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_YaleSTC_reservations_pull_1612-23discussion-5Fr76820050&d=CwMFaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=fGUlyXMXUllcq1Lzipj1Qfu_NL-lk9lC6GgNV_p2VOg&m=eBycs-uApYBSkl1DeskJyUjpxLMk08qdahucufR-eHs&s=dODIql38lSBpgd7EvEoIhHDjOfPh1iDSNjOKp70Deg8&e=>
> :
>
> >      end
> >
> >      context 'when accessed by checkout person disallowed by settings' do
> >        before(:each) do
> > -        sign_in @checkout_person
> > -        allow(@ac).to receive(:checkout_persons_can_edit).and_return(false)
> > -        get 'edit', id: @reservation.id
>
> Well, here we don't actually have a reservation created because it should
> redirect before trying to get the reservation in question.
>
> get :edit requires an id to start the request, so I just used 1.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_YaleSTC_reservations_pull_1612_files_3ff912785e3867da977312ee971b34c6731fe134-23r76820050&d=CwMFaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=fGUlyXMXUllcq1Lzipj1Qfu_NL-lk9lC6GgNV_p2VOg&m=eBycs-uApYBSkl1DeskJyUjpxLMk08qdahucufR-eHs&s=ptT_uMcKgio2l16iBqQBmdtdjnVT6GL2CCwQH-p0ekM&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AHVWMOMEoVFk-5Fv34QqxoGYWEmoMOalbjks5qlE14gaJpZM4JO8Vp&d=CwMFaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=fGUlyXMXUllcq1Lzipj1Qfu_NL-lk9lC6GgNV_p2VOg&m=eBycs-uApYBSkl1DeskJyUjpxLMk08qdahucufR-eHs&s=8AW9wl3G0PvVo_02Du3N-JJqyaNkRpvY9Oq9OSNuPRk&e=>
> .
>